商品一覧表示

K-Hir-pk18 opened the pull request:
WHAT
商品一覧表示

WHY
閲覧者が出品商品を見れるように

K-Hir-pk18 committed:
商品一覧表示

WHAT
商品一覧表示

WHY
閲覧者が出品商品を見れるように

takahirohoriguchi commented:
レビュー依頼ありがとうございます！
こちらのプルリクエストですが、実際の画面や機能ができているか確認させていただきたいので、下記キャプチャ動画をご提出いただけますでしょうか？

・トップページに出品した商品が表示されている動画
・商品のデータがない場合に、ダミー商品が表示されている動画

その後、再度レビュー依頼フォームにてご提出をお願いいたします。

K-Hir-pk18 commented:
takahirohoriguchi様

ご対応ありがとうございます。
以下にキャプチャを載せておきます。

・トップページに出品した商品が表示されている動画
https://gyazo.com/cdd638177d83766ef08468c8d5902f0f

・商品のデータがない場合に、ダミー商品が表示されている動画
https://gyazo.com/be10ec8dacdd054d1990d2cf94e9ad53

K-Hir-pk18 commented:
ご担当者様

よろしくお願いいたします。

masayaomoto commented on app/controllers/products_controller.rb:
```diff
@@ -5,6 +5,7 @@ class ProductsController < ApplicationController
 
   def index
     @products = Product.all
```
```ruby
@products = Product.all
```
こちら、商品を全て表示するだけでなく
orderメソッドを用いて出品した順に並び替えましょう！

masayaomoto commented on app/controllers/products_controller.rb:
```diff
@@ -5,6 +5,7 @@ class ProductsController < ApplicationController
 
   def index
     @products = Product.all
+    #@purchases = Purchase.all
```
```
#@purchases = Purchase.all
```
こちらのコメントアウトですが、不要であれば削除しましょう。
もし必要であれば、第三者でもわかるように「残す目的」「なぜ必要か」という理由などを、コメントとして書いておくことをおすすめいたします。

K-Hir-pk18 replied:
masayaomoto様

ご対応ありがとうございます。
変更しました。

masayaomoto reviewed (commented):
レビュー依頼ありがとうございます！
キャプチャ動画確認させていただきました。
追加でコメントいたしましたので、ご確認お願いします！

K-Hir-pk18 committed:
商品一覧表示　レビュー対応

WHAT
コメント対応
　・controllerのコメントアウト消去
　　　→あとで使うと確信持てず
　・indexのcontroller
　　　→orderメソッドの追記

WHY
実装に必要な機能のため

K-Hir-pk18 commented:
ご担当者様

よろしくお願いいたします。

takahirohoriguchi commented on app/controllers/products_controller.rb:
```diff
@@ -4,7 +4,7 @@ class ProductsController < ApplicationController
 
   def index
-    @products = Product.all
+    @products = Product.all.order("created_at ASC")
```
```
    @products = Product.all.order("created_at ASC")
```
こちら昇順になっているため、古い商品から表示されるようになっています。
そのため、新しい商品から表示されるように`ASC`から`DESC`へ修正しましょう。

takahirohoriguchi commented on app/views/products/index.html.erb:
```diff
@@ -122,65 +122,65 @@
-            </div>
-          </div>
-        </div>
+        <%# //商品がない場合は以下のダミー商品が表示されるようにしましょう %>
+        <% if @products.count == 0 %>
```
こちらのコメントアウトですが、不要であれば削除しましょう。
もし必要であれば、第三者でもわかるように「残す目的」「なぜ必要か」という理由などを、コメントとして書いておくことをおすすめします。

K-Hir-pk18 replied:
takahirohoriguchi様

レビューありがとうございます。
がんばります✨✨

takahirohoriguchi reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました！
2点コメントしましたので、ご確認をお願いいたします！あと少しです！
ご不明点ございましたらコネクトにて相談していただけると幸いです。

K-Hir-pk18 committed:
商品一覧表示　レビュー対応２

WHAT
コメント対応
　・product contoroller
　　→ASCからDESCへ修正

　・products/indexのコメントアウト
　　→削除

WHY

K-Hir-pk18 commented:
ご担当者様

よろしくお願いいたします。

takahirohoriguchi commented:
> レビューありがとうございます。
> がんばります✨✨

コメントありがとうございます！応援しています！！

レビュー依頼ありがとうございます！
LGTMとさせていただきます！実装お疲れ様でした！
次は商品詳細機能ですね！引き続き実装頑張っていきましょう！

K-Hir-pk18 commented:
takahirohoriguchi様

レビューありがとうございます！
コメントも嬉しいです😆

LGTMもいただいたのでさらに頑張っていきたいです！！
